Tools/ardupilotwaf/boards.py

```diff
@@ -44,7 +44,7 @@ def configure(self, cfg):
                     keys.sort()
                 val = ['%s=%s' % (vk, val[vk]) for vk in keys]
 
-            if k in cfg.env and isinstance(cfg.env, list):
+            if k in cfg.env and isinstance(cfg.env.k, list):
```

Comment:
No, it's not. With `cfg.env.k` the value of `name` in `__getattr__` is `'k'`. Now, `cfg.env[k]` triggers a call like `cfg.env.__getitem__(k)`.

In other words, in `cfg.env.k`, 'k' is used as key (or attribute name), whereas in `cfg.env[k]`, the value of `k` is used instead.